When user setting is nil, don't init loading progress bar

bryce-carson opened the pull request:
When the user customizes `dotspacemacs-loading-progress-bar` and sets it to nil, do not initialize the loading progress bar during startup.

This prevents the full progress bar from overtaking the mode-line, preventing redisplays that are unnecessary when no progress bar updates will occur later for visual fluff.

In short, don't show an empty progress bar if we don't intend to fill it.

bryce-carson committed:
When user setting is nil, don't init loading progress bar

When the user customizes `dotspacemacs-loading-progress-bar` and sets it to nil,
do not initialize the loading progress bar during startup.

This prevents the full progress bar from overtaking the mode-line, preventing
redisplays that are unnecessary when no progress bar updates will occur later
for visual fluff.

In short, don't show an empty progress bar if we don't intend to fill it.

lebensterben commented on core/core-configuration-layer.el:
```diff
@@ -2089,7 +2089,7 @@ LAYER must not be the owner of PKG."
 (defun configuration-layer//configure-package (pkg)
   "Configure PKG object, i.e. call its post-init function."
-  (spacemacs/update-progress-bar)
+  (when dotspacemacs-loading-progress-bar (spacemacs/update-progress-bar))
```
this is unnecessary 

https://github.com/syl20bnr/spacemacs/blob/0bbb41ec97198b19e1e44ee604caf56596506332/core/core-progress-bar.el#L53

lebensterben commented on core/core-configuration-layer.el:
```diff
@@ -1954,7 +1954,7 @@ RNAME is the name symbol of another existing layer."
 (defun configuration-layer//configure-packages (packages)
   "Configure all passed PACKAGES honoring the steps order."
-  (spacemacs/init-progress-bar (length packages))
+  (when dotspacemacs-loading-progress-bar (spacemacs/init-progress-bar (length packages)))
```
can be added here instead 

https://github.com/syl20bnr/spacemacs/blob/0bbb41ec97198b19e1e44ee604caf56596506332/core/core-progress-bar.el#L35

bryce-carson committed:
Wrap the init function body in an `when`

Resolves https://github.com/syl20bnr/spacemacs/pull/15997#discussion_r1181115472.

bryce-carson committed:
Revert "When user setting is nil, don't init loading progress bar"

This reverts commit b244f101b0d69d80b131075986e1ea3871ae40f5.

lebensterben reviewed (commented):
could you also change the docstring of update-progress-bar and init-progress-bar to mention that they have no effect when dotspacemacs-loading-progress-bar is nil?

bryce-carson commented:
> could you also change the docstring of update-progress-bar and init-progress-bar to mention that they have no effect when dotspacemacs-loading-progress-bar is nil?

Absolutely. I should've done that without being reminded. Thanks.

bryce-carson committed:
Update function documentation.

bryce-carson committed:
Make the variables display in the quoted form.

bryce-carson committed:
Fix quotation issue.

lebensterben commented on core/core-progress-bar.el:
```diff
@@ -46,8 +48,11 @@
 (defun spacemacs/update-progress-bar ()
-  "Update progress bar by incrementing its value by 1.
-Display the progress bar by chunks of size `spacemacs-loading-dots-chunk-threshold'"
+  "When the customization variable
```
the first sentence of the docstring should be less than 80 chars.

lebensterben commented on core/core-progress-bar.el:
```diff
@@ -46,8 +48,11 @@
 (defun spacemacs/update-progress-bar ()
-  "Update progress bar by incrementing its value by 1.
-Display the progress bar by chunks of size `spacemacs-loading-dots-chunk-threshold'"
+  "When the customization variable
```
The first sentence should end in a period.

Just use the original docstring. And add a new line "This has no effect when xxx is nil."

bryce-carson replied:
I would've reverted to the original docstring if I saw that earlier. The new string should be okay, however.

bryce-carson committed:
Fix documentation string length. Further improve wording.

lebensterben commented on core/core-progress-bar.el:
```diff
@@ -33,7 +33,11 @@
 
 (defun spacemacs/init-progress-bar (max)
-  "Initialize the progress bar."
+  "Conditionally initialize the package-loading progress bar.
```
```suggestion
  "Initialize the package-loading progress bar.
```

lebensterben commented on core/core-progress-bar.el:
```diff
@@ -33,7 +33,11 @@
 (defun spacemacs/init-progress-bar (max)
-  "Initialize the progress bar."
+  "Conditionally initialize the package-loading progress bar.
+
```
```suggestion
```

lebensterben commented on core/core-progress-bar.el:
```diff
@@ -33,7 +33,11 @@
-  "Initialize the progress bar."
+  "Conditionally initialize the package-loading progress bar.
+
+When the customziation variable `dotspacemacs-loading-progress-bar' is non-nil,
```
Can you just do what a said?

bryce-carson replied:
Do I still need to do anything after your new commits? There are still new lines indicating the no-op when the variable is `nil`.

At the moment my Emacs is crashing because I listened to the warning coming from Magit #15992 .

lebensterben commented on core/core-progress-bar.el:
```diff
@@ -46,8 +50,13 @@
 (defun spacemacs/update-progress-bar ()
-  "Update progress bar by incrementing its value by 1.
-Display the progress bar by chunks of size `spacemacs-loading-dots-chunk-threshold'"
+  "Conditionally update the package-loading progress bar.
```
```suggestion
  "Update the package-loading progress bar.
```

lebensterben commented on core/core-progress-bar.el:
```diff
@@ -46,8 +50,13 @@
-  "Update progress bar by incrementing its value by 1.
-Display the progress bar by chunks of size `spacemacs-loading-dots-chunk-threshold'"
+  "Conditionally update the package-loading progress bar.
+
```
```suggestion
```

bryce-carson committed:
Revert to original docstring, including "does nothing" statement.

Also includes a missing period after the original docstring in the
`spacemacs/update-progress-bar` function.

bryce-carson commented:
Fuck. I have no idea what is going on in this branch anymore. I'm pretty confused with Git at the moment.

bryce-carson commented:
@lebensterben I'm deleting this branch and PR because Git was very confusing on my end and I wasn't able to push the proper changes to this broken branch to make the request changes.

See the new PR #16000.
